### ISSUE TITLE
Spaces and other special characters in the file path may break sourcemaps

### ISSUE THREAD

**Issue by Gabriel0402** [Microsoft/vscode-chrome-debug#64]
I'm building an application with angular2 alpha 48, typescript and webpack. Thanks for the effort so that I can debug inside vscode(before this I could only debug inside chrome developer debugger). I have a configuration which compile the ts files from private folder to an app.js which in the public folder which would be hosted under the website root.

The structure may look like this:
```
 ├──src/                                   * 
 │   │
 │   ├──app/                               * private folder host the whole ts files
 │   │   ├──app.ts                         * App.ts: app component 
 │   │   
 │   │
 │   ├──public/                            * 
 │   │   ├──app.js                           * compiled js file from ts file
 │   │   ├──app.js.map                   * source map
```
I could set breakpoints in app.js and I'm able to debug with it. And in the log I saw that "SourceMaps are enabled but app.js is a runtime script." And app.js.map maps to the app.ts with absolute path inside. But I'm not able to set the breakpoint inside app.ts. "The log is  .../app.ts cannot be resolved to a loaded script." I'm wondering whether this is caused by the runtime script. Is it because that app.js is runtime script so that sourcemap is not working in this case. Is there any future plan for this? Thanks!

And one update is that chrome debugger is able to map the file to the correct ts files.

**Comment by roblourens** [comment]
It's interesting that it gives '../app.ts' in that message, because I'm expecting that path to be an absolute path, not a relative path. That would definitely be the problem, and I'm not sure why Code is giving a relative path there. Thanks for the report!

Is public/ the folder that you have open in Code? You might want to try it with 'src/' as the folder open in Code.

**Comment by Gabriel0402** [comment]
Sorry my fault, the ../app.ts is absolute path. So it's c:\dev\myproject\src\app\app.ts. I open the folder in myproject level and set the webroot to src/public. I think this works fine as I'm able to debug inside the app.js. The problem is that "c:\dev\myproject\src\app\app.ts" is not the one in sourcemap. In sourcemap the path is "file:///C:\\dev\\myproject\\src\\app\\app.ts". My question is whether the extension could resolve the path in sourcemap to the path:"c:\dev\myproject\src\app\app.ts" which I open in vscode.

**Comment by roblourens** [comment]
I see. The message "app.js is a runtime script" just means that sourcemaps are enabled, but a breakpoint is set in the generated file, not the TS file. That's ok, but line numbers may get messed up if you try to debug this way without turning off sourcemaps.

Yeah, using file:/// urls in your sourcemap should be resolved correctly. I can't tell what's wrong, but if you post the full log I may be able to help you debug.

**Comment by Gabriel0402** [comment]
Thanks a lot for your help! Really appreciate it. 
I've attached the log. Here're some more details. I also included it in the bottom of the log.
When I add break point inside app.js, here's the log:
```
›From client: setBreakpoints({"source":{"path":"c:\\Users\\czhu\\Desktop\\MyProject\\src\\public\\__build__\\app.js"},"lines":[5436,5492]})
  ›SourceMaps.setBP: SourceMaps are enabled but c:\Users\czhu\Desktop\MyProject\src\public\__build__\app.js is a runtime script
  ›Paths.setBP: Resolved c:\Users\czhu\Desktop\MyProject\src\public\__build__\app.js to https://localhost:3000/__build__/app.js
  ›To target: {"id":11,"method":"Debugger.removeBreakpoint","params":{"breakpointId":"https://localhost:3000/__build__/app.js:5435:0"}}
  ›From target: {"id":11,"result":{}}
  ›To target: {"id":12,"method":"Debugger.setBreakpointByUrl","params":{"url":"https://localhost:3000/__build__/app.js","lineNumber":5435,"columnNumber":0}}
  ›To target: {"id":13,"method":"Debugger.setBreakpointByUrl","params":{"url":"https://localhost:3000/__build__/app.js","lineNumber":5491,"columnNumber":0}}
  ›From target: {"id":12,"result":{"breakpointId":"https://localhost:3000/__build__/app.js:5435:0","locations":[{"scriptId":"6534","lineNumber":5435,"columnNumber":9}]}}
  ›From target: {"id":13,"result":{"breakpointId":"https://localhost:3000/__build__/app.js:5491:0","locations":[{"scriptId":"6534","lineNumber":5491,"columnNumber":13}]}}
  ›To client: {"seq":0,"type":"response","request_seq":8,"command":"setBreakpoints","success":true,"body":{"breakpoints":[{"verified":true,"line":5436},{"verified":true,"line":5492}]}}
```
When I add breakpoint inside aside.component.ts, here's the log:
```
 ›From client: setBreakpoints({"source":{"path":"c:\\Users\\czhu\\Desktop\\MyProject\\src\\app\\components\\display\\aside.component.ts"},"lines":[617,286,361]})
  ›SourceMaps.setBP: c:\Users\czhu\Desktop\MyProject\src\app\components\display\aside.component.ts can't be resolved to a loaded script.
```
[log.txt](https://github.com/Microsoft/vscode-chrome-debug/files/58818/log.txt)

**Comment by roblourens** [comment]
So the sourcemapped scripts are getting resolved to a path like `c:\\Users\\czhu\\Desktop\\PI%20Nova\\src\\app\\components\\display\\aside.component.ts`, but the actual location of the script is like `c:\\Users\\czhu\\Desktop\\MyProject\\src\\app\\components\\display\\aside.component.ts`. I don't see any `PI%20Nova` in the sourcemap you posted, so any idea where that's coming from? Did you build the project in one folder and move it to another?

**Comment by roblourens** [comment]
By the way, this question is you right? http://stackoverflow.com/questions/34185748/how-to-make-webpack-sourcemap-to-original-files just want to make sure I'm getting to everyone.

**Comment by Gabriel0402** [comment]
Sorry I create a new project for test purpose. Something messed up there I suppose. I've created a new log from the original project. In the log it seems that app.js has mapped sources but with PI%20Nova. But the resolved path is PI Nova. I'm wondering whether this cause the problem? And this question is mine, Thanks a lot for your help. Really appreciate it.
[log.txt](https://github.com/Microsoft/vscode-chrome-debug/files/59715/log.txt)

**Comment by Gabriel0402** [comment]
Oh I got it! It's the space that cause this problem! So the project name should have no space. After I get rid of the space I could set the break point in the ts file correctly. Thanks a lot for your help! I'm not sure whether there's any plans for the extension to parse the space? If not I'll just change the project name.  Thanks again for your kindness!

**Comment by roblourens** [comment]
Oh yeah, that is the problem. I need to urlencode one end of that comparison. Some other characters will probably break it too. Thanks for helping me track it down!

**Comment by sumigoma** [comment]
+1

**Comment by roblourens** [comment]
This is fixed in v0.2.1

**Comment by Gabriel0402** [comment]
Thanks, it works now!
